server/src/test/java/org/elasticsearch/search/aggregations/metrics/CardinalityAggregatorTests.java

```diff
@@ -112,6 +213,180 @@ public void testQueryFiltersAll() throws IOException {
+            iw.addDocument(singleton(new SortedDocValuesField("str_value", new BytesRef("one"))));
+        }, card -> {
+            assertEquals(2, card.getValue(), 0);
+            assertTrue(AggregationInspectionHelper.hasValue(card));
+        }, mappedFieldTypes);
+    }
+
+    public void testSingleValuedStringScript() throws IOException {
+        final CardinalityAggregationBuilder aggregationBuilder = new CardinalityAggregationBuilder("name").script(
+            new Script(ScriptType.INLINE, MockScriptEngine.NAME, "doc['str_value'].value", emptyMap())
+        );
+        final MappedFieldType mappedFieldTypes = new KeywordFieldMapper.KeywordFieldType("str_value");
+
+        testAggregation(aggregationBuilder, new MatchAllDocsQuery(), iw -> {
+            iw.addDocument(singleton(new SortedDocValuesField("str_value", new BytesRef("one"))));
+            iw.addDocument(singleton(new SortedDocValuesField("unrelatedField", new BytesRef("two"))));
+            iw.addDocument(singleton(new SortedDocValuesField("str_value", new BytesRef("three"))));
+            iw.addDocument(singleton(new SortedDocValuesField("str_value", new BytesRef("one"))));
+        }, card -> {
+            assertEquals(2, card.getValue(), 0);
+            assertTrue(AggregationInspectionHelper.hasValue(card));
+        }, mappedFieldTypes);
+    }
+
+    public void testMultiValuedStringScript() throws IOException {
+        final CardinalityAggregationBuilder aggregationBuilder = new CardinalityAggregationBuilder("name").script(
+            new Script(ScriptType.INLINE, MockScriptEngine.NAME, "doc['str_values']", emptyMap())
+        );
+        final MappedFieldType mappedFieldTypes = new KeywordFieldMapper.KeywordFieldType("str_values");
+
+        testAggregation(aggregationBuilder, new MatchAllDocsQuery(), iw -> {
+            iw.addDocument(
+                Arrays.asList(
+                    new SortedSetDocValuesField("str_values", new BytesRef("one")),
+                    new SortedSetDocValuesField("str_values", new BytesRef("two"))
+                )
+            );
+            iw.addDocument(
+                Arrays.asList(
+                    new SortedSetDocValuesField("unrelatedField", new BytesRef("two")),
+                    new SortedSetDocValuesField("unrelatedField", new BytesRef("three"))
+                )
+            );
+            iw.addDocument(
+                Arrays.asList(
+                    new SortedSetDocValuesField("str_values", new BytesRef("two")),
+                    new SortedSetDocValuesField("str_values", new BytesRef("three"))
+                )
+            );
+            iw.addDocument(
+                Arrays.asList(
+                    new SortedSetDocValuesField("str_values", new BytesRef("one")),
+                    new SortedSetDocValuesField("str_values", new BytesRef("three"))
+                )
+            );
+            iw.addDocument(
+                Arrays.asList(
```

Comment:
It's not important, but in more modern code we've moved to `List.of` here. The list is makes is immutable which is nice. It doesn't make a difference here, but it's still nice.